File: Java/[Udemy] Java POO/section13/HerancaEx1/src/entities/OutsourcedEmployee.java
package entities;

public class OutsourcedEmployee extends Employee {

	private Double additionalCharge;
	
	public OutsourcedEmployee(String name, Integer hours, Double valuePerHour, Double additionalCharge) {
		super(name, hours, valuePerHour);
		this.additionalCharge = additionalCharge;
	}
	
	public Double getAdditionalCharge() {
        return additionalCharge;
    }
	
	public void setAdditionalCharge(Double additionalCharge) {
        this.additionalCharge = additionalCharge;
    }
	
	@Override
	public final Double payment() {
        return super.payment() + getAdditionalCharge() * 1.1;
    }
	
	@Override
	public String toString() {
        return String.format("%s - $%.2f", super.getName(), payment());
    }
}

File: CSharp/[Udemy] CSharp POO/session8/Session100Ex2/Entities/OutsourcedEmployee.cs
﻿namespace Session100Ex2.Entities
{
    internal class OutsourcedEmployee : Employee
    {
        public double AdditionalCharge { get; set; }

        public OutsourcedEmployee() { }
        public OutsourcedEmployee(string name, int hours, double valuePerHour, double additionalCharge) : base (name, hours, valuePerHour)
        {
            AdditionalCharge = additionalCharge;
        }
        public override double Payment()
        {
            return base.Payment() + (AdditionalCharge * 1.10);
        }
    }
}